codec-http/src/main/java/io/netty/handler/codec/http/QueryStringDecoder.java

```diff
@@ -196,7 +191,7 @@ public String path() {
      */
     public Map<String, List<String>> parameters() {
         if (params == null) {
-            params = decodeParams(uri, pathEndIdx(), charset, maxParams, semicolonIsNormalChar);
+            params = decodeParams(uri, pathEndIdx(), charset, null, maxParams, semicolonIsNormalChar);
```

Comment:
We could also add a variant of this method that returned void and took the callback as a parameter.

Reply:
That's indeed a very good idea!